core/src/nucleus/mod.rs

```diff
@@ -251,17 +262,14 @@ pub fn reduce(
                                         &observer_channel,
                                     );
 
-                                    // genesis returns a i32 as a string
-                                    // 0 == success
+                                    // genesis returns a string
+                                    // "" == success, otherwise error value
                                     match call_result {
-                                        // not okay if genesis returned an errorCode
-                                        Ok(ref s) if s != "0" => {
+                                        // not okay if genesis returned an value
+                                        Ok(ref s) if s != "" => {
```

Comment:
is this where we should use an enum to represent the possible return strings and what the error codes might be?

Reply:
I don't think so because these may be application specific.  There's no predefined set.  Of course there may be some pre-defined values so we could go with and ErrorGeneric as we did in the errors.rs

Reply:
that makes sense to me

i'm imagining something like this https://doc.rust-lang.org/std/io/enum.ErrorKind.html

that has an `Other` and you also provide a text message to fit your type